### PR TITLE
Remove obsoleted methods

### PR DESCRIPTION
**Pull request by igitur**
Remove obsoleted methods from #1171 and #1029

### CODE REVIEW

File: ClosedXML/Excel/Ranges/IXLRangeBase.cs [hunk 1]
```diff
@@ -118,20 +109,18 @@ public interface IXLRangeBase : IXLAddressable
         IXLCell FirstCellUsed();
 
         /// <summary>
-        ///   Returns the first cell with a value of this range.
+        /// Returns the last cell with a value of this range.
```
Comment (Pankraty) [inline ClosedXML/Excel/Ranges/IXLRangeBase.cs]: Looks like this should be `first`
```suggestion
        /// Returns the first cell with a value of this range.
```

File: ClosedXML/Excel/Ranges/IXLRangeBase.cs [hunk 2]
```diff
@@ -118,20 +109,18 @@ public interface IXLRangeBase : IXLAddressable
...
+        /// <param name="options">The options to determine whether a cell is used.</param>
         IXLCell FirstCellUsed(XLCellsUsedOptions options);
 
         IXLCell FirstCellUsed(Func<IXLCell, Boolean> predicate);
 
-        [Obsolete("Use the overload with XLCellsUsedOptions")]
-        IXLCell FirstCellUsed(Boolean includeFormats, Func<IXLCell, Boolean> predicate);
-
+        /// <summary>
+        /// Returns the last cell with a value of this range.
```
Comment (Pankraty) [inline ClosedXML/Excel/Ranges/IXLRangeBase.cs]: Same here

File: ClosedXML/Excel/Ranges/IXLRangeBase.cs (hunk 1 above)
Comment (Pankraty) [reply ClosedXML/Excel/Ranges/IXLRangeBase.cs]: Also, the comment may be a bit misleading - cells considered non-empty does not necessarily have a value. Maybe 
```suggestion
        /// Returns the first non-empty cell of this range.
```
?

File: ClosedXML.Tests/Excel/Misc/XLWorkbookTests.cs [hunk 3]
```diff
@@ -359,7 +359,7 @@ public void WbProtect2()
             using (var wb = new XLWorkbook())
             {
                 var ws = wb.Worksheets.Add("Sheet1");
-                wb.Protect(true, false);
```
Comment (jahav) [inline ClosedXML.Tests/Excel/Misc/XLWorkbookTests.cs:362]: The protection API is rather confusing API. The original parameters true and false were for lockStructure and lockWindows.... but the Protect(XLWorkbookProtectionElements.Windows) indicates which elements are allowed... I would expect it to be opposite.

* `wb.Protect(XLWorkbookProtectionElements.None)` means "nothing is allowed to changes", but it reads as protect no element
* `wb.Protect(XLWorkbookProtectionElements.Everything)` means "everything is allowed to changes", but it reads as protect everything.

**Review by jahav — APPROVED** [review]
It is not directly related with the PR, but the API for protection is inversion of what would be expected.

**Pull request merged**